Shared/Samples/Animate images with image overlay/AnimateImagesWithImageOverlayView.swift

```diff
@@ -0,0 +1,190 @@
+            let images = imageURLs
+                .sorted { $0.lastPathComponent < $1.lastPathComponent }
+                .map { UIImage(contentsOfFile: $0.path)! }
+            return CircularIterator(elements: images)
+        }()
+        
+        /// A formatter to format percentage strings.
+        let percentageFormatter: NumberFormatter = {
+            let formatter = NumberFormatter()
+            formatter.numberStyle = .percent
+            formatter.multiplier = 100
+            return formatter
+        }()
+        
+        /// The frame rate speed of the image overlay
+        @Published var fpsSpeed = 30
+        
+        /// The image overlay to show image frames.
+        @Published var imageOverlay: ImageOverlay = {
+            let imageOverlay = ImageOverlay()
+            imageOverlay.opacity = 0.5
+            return imageOverlay
+        }()
+        
+        init() {
+            // Create new display link.
+            let newDisplayLink = CADisplayLink(target: self, selector: #selector(setImageFrame))
+            
+            // Inherit the frame rate from existing display link, or set to default 60 fps.
+            newDisplayLink.preferredFramesPerSecond = displayLink?.preferredFramesPerSecond ?? 60
+            newDisplayLink.isPaused = true
+            
+            // Add to main thread common mode run loop, so it is not effected by UI events.
+            newDisplayLink.add(to: .main, forMode: .common)
+            displayLink = newDisplayLink
+        }
+        
+        /// Set the image frame to the next one.
+        @objc
+        func setImageFrame() {
+            if let image = imagesIterator.next() {
+                let frame = ImageFrame(image: image, extent: .pacificSouthwestExtent)
+                imageOverlay.imageFrame = frame
+            }
+        }
+    }
+    
+    /// A generic circular iterator.
+    private struct CircularIterator<Element>: IteratorProtocol {
+        /// An array of elements to be iterated over.
+        let elements: [Element]
+        
+        /// The element iterator.
+        private var elementIterator: Array<Element>.Iterator
+        
+        init(elements: [Element]) {
+            self.elements = elements
+            elementIterator = elements.makeIterator()
+        }
+        
+        /// Moves to the next element if there is one or starts over by creating
+        /// a new iterator.
+        /// - Returns: The next element.
+        mutating func next() -> Element? {
+            if let next = elementIterator.next() {
+                return next
+            } else {
+                elementIterator = elements.makeIterator()
+                return elementIterator.next()
+            }
+        }
+    }
+}
+
+private extension Envelope {
+    /// An envelope of the pacific southwest sector for displaying the image frame.
+    static var pacificSouthwestExtent = Envelope(
```

Comment:
Since the `ImageFrame` is constantly updated, there is a small performance overhead to recalculate the envelope on every frame. Thus I recommend putting it back to a instance property.